### DISCUSSION TITLE
0.38.0

### DISCUSSION THREAD

**Discussion by rmorshea** [idom-team/idom#735] [Announcements]
## What's Changed
* Rework How IDOM Integrates With Servers by @rmorshea in https://github.com/idom-team/idom/pull/703
* Implement use_location by @rmorshea in https://github.com/idom-team/idom/pull/721
* Use "backend" instead of "server" by @rmorshea in https://github.com/idom-team/idom/pull/726
* make Layout context management async by @rmorshea in https://github.com/idom-team/idom/pull/730
* Improve Changelog by @rmorshea in https://github.com/idom-team/idom/pull/731
* add use debug value hook by @rmorshea in https://github.com/idom-team/idom/pull/733

**Full Changelog**: https://github.com/idom-team/idom/compare/0.37.2...0.38.0

<hr /><em>This discussion was created from the release <a href='https://github.com/idom-team/idom/releases/tag/0.38.0'>0.38.0</a>.</em>

**Comment by rmorshea** [comment]
@acivitillo @Archmonger, outside of critical bug-fixes that arise in the next week, this is the last release before my vacation.

**Reply by Archmonger** [reply]
Can we push out a Django-IDOM version bump before your vacation?

**Reply by rmorshea** [reply]
Sure.

**Comment by acivitillo** [comment]
we will build on top of the current version and will contribute on some documentation to idom

**Reply by Archmonger** [reply]
Feel free to request for my review on any docs.

**Reply by rmorshea** [reply]
@acivitillo all the documentation I've been writing has been closely following the new ReactJS docs (https://beta.reactjs.org/) in structure and content. The documentation they're creating there is **absolutely incredible** and because they're using a [creative commons license](https://idom-docs.herokuapp.com/docs/about/credits-and-licenses.html) we can copy text and examples directly where applicable. While writing IDOM's current documentation was a momentous effort (3 months), I have to imagine it would have taken a year or more without the ability to copy and take inspiration from React's docs.

When you contribute documentation, it would be ideal if you could first take a look at beta.reactjs.org to see if there's a section of documentation they've already written on the subject. If there is, just take that section and regurgitate it in a form that applies to IDOM. Obviously you should feel free to improve on what React has done, but copying examples and the structure of the docs has really sped things up for me. An example of a section I think I was able to improve was [Introduction to use_state](https://idom-docs.herokuapp.com/docs/guides/adding-interactivity/components-with-state/index.html#introduction-to-use-state) vs ReactJS's [Anatomy of useState](https://beta.reactjs.org/learn/state-a-components-memory#anatomy-of-usestate)

**Reply by acivitillo** [reply]
Ok will follow react docs for standard documentation. However, there are patterns I need documented and I am happy to contribute them back to idom if it helps. To give you am example, it took a while to figure out the best way to auth with idom is via embedded app, also the pattern of pages as apps is another one that needs research and when best practices are established documentation. Currently I am thinking on writing a tutotial on how to auth with fastapi using a github idp and passing current_user to embedded idom.

**Reply by Archmonger** [reply]
We are going to be reworking how IDOM is used in production when resolving this issue:
- #653 

IDOM will primarily exist as [an embeddable template tag](https://github.com/idom-team/django-idom#example_apptemplatesyour-templatehtml), identical to Django IDOM. The docs will be refactored to push this narrative. So hopefully, this change on it's own will make it easier to understand how/when to integrate with frameworks.

I don't know if extremely specialized tutorials, like FastAPI authentication, belong in our core docs. I think it's good for us to have some degree tutorials, but the core docs would get flooded if we started hosting things like that. Maybe we should consider some kind of wiki-like structure for this.
